src/TypedMap.php

```diff
@@ -163,15 +183,14 @@ public function getValues()
      */
     public function has($key)
     {
-        return is_object($key) ?
-            in_array($key, $this->keys) :
-            array_key_exists($key, $this);
+        return array_key_exists($this->getKeyId($key), $this->keys);
     }
 
     /**
      */
     public function clear()
```

Comment:
Just thinking, why would we allow to clear this map object instead of forcing to create a new one?